storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -1988,8 +1988,12 @@ static void copy_back_thread_func(datadir_thread_ctxt_t *ctx) {
 
     std::string dst_path = entry.rel_path;
 
-    if (file_purpose == FILE_PURPOSE_DATAFILE ||
-        file_purpose == FILE_PURPOSE_UNDO_LOG) {
+    if (file_purpose == FILE_PURPOSE_UNDO_LOG) {
+      /* undo tablespace can only be in undo_dir or data dir */
+      std::string dst_dir =
+          (srv_undo_dir && *srv_undo_dir) ? srv_undo_dir : mysql_data_home;
+      dst_path = dst_dir + "/" + dst_path;
```

Comment:
what is dst_path here? why do we need it?
 srv_undo_dir is directory specified for prepare, so this is the place where undo files should be put. Didn't get the purpose of ds_path

Reply:
Initially, dst_path has file_name of undo_tablespace. 
and we prefix it with either with srver_undo_dir or mysql_data_dir.

Reply:
Ok just the filename? wihout any paths? Or can it contain "./"

Reply:
just file_name without any paths.